specs/experimental/op-stack-manager.md

```diff
@@ -0,0 +1,214 @@
+[contract releases]: https://github.com/ethereum-optimism/optimism/blob/main/packages/contracts-bedrock/VERSIONING.md
+[standard configuration]: ../protocol/configurability.md
+[superchain registry]: https://github.com/ethereum-optimism/superchain-registry
+[ethereum-lists/chains]: https://github.com/ethereum-lists/chains
+[Batch Inbox]: ../protocol/configurability.md#consensus-parameters
+
+The OP Stack Manager is a contract that deploys the L1 contracts for an OP Stack chain in a single
+transaction. It provides a minimal set of user-configurable parameters to ensure that the resulting
+chain meets the [standard configuration] requirements.
+
+The version deployed is always a governance-approved contract release. The set
+of governance approved [contract releases] can be found on the
+[Optimism Monorepo releases] page, and is the set of releases named
+`op-contracts/vX.Y.Z`.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Deployment](#deployment)
+- [Interface](#interface)
+  - [`Proxy.sol`](#proxysol)
+  - [`deploy`](#deploy)
+  - [Getter Methods](#getter-methods)
+- [Implementation](#implementation)
+  - [Batch Inbox Address](#batch-inbox-address)
+  - [Contract Deployments](#contract-deployments)
+- [Security Considerations](#security-considerations)
+  - [Chain ID Source of Truth](#chain-id-source-of-truth)
+  - [Chain ID Frontrunning](#chain-id-frontrunning)
+  - [Chain ID Value](#chain-id-value)
+  - [Proxy Admin Owner](#proxy-admin-owner)
+  - [Upgradeability (ABI Changes)](#upgradeability-abi-changes)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Deployment
+
+The OP Stack Manager is a proxied contract deployed at `0xTODO`. It can be deployed as follows:
+
+TODO.
+
+## Interface
+
+Version 1.0.0 of the OP Stack Manager deploys the [`op-contracts/v1.4.0`]
+contracts release.
+
+### `Proxy.sol`
+
+The OP Stack Manager is a proxied contract using the standard `Proxy.sol` contract that lives in
+the Optimism monorepo. Therefore the OP Stack Manager will have the same interface as the
+`Proxy.sol`, in addition to other methods defined in this specification.
+
+The privileged methods of the OP Stack Manager will be held by the L1 ProxyAdmin owner, as
+specified by the [standard configuration].
+
+### `deploy`
+
+The `deploy` method is the only non-view method in the contract. It is used to
+deploy the full set of L1 contracts required to setup a new OP Stack chain that
+complies with the [standard configuration]. It has the following interface:
+
+```solidity
+struct Roles {
+  address proxyAdminOwner;
+  address systemConfigOwner;
+  bytes32 batcherHash;
+  address unsafeBlockSigner;
+  address proposer;
+  address challenger;
+}
+
+function deploy(
+  uint64 l2ChainId,
+  Roles roles,
+  uint256 scalar
```

Comment:
This is using the pre ecotone `SystemConfig`?

Reply:
The overhead should always be 0 and I recommend we give a nice human readable way to set the base fee scalar and blob fee scalar and then serialize it as the `scalar` on chain

Reply:
Nice UX suggestions, updated in be28df1